pkg/fleet/installer/service/apm_inject.go

```diff
@@ -80,61 +81,64 @@ func (a *apmInjectorInstaller) Setup(ctx context.Context) (err error) {
 		return err
 	}
 
-	rollbackLDPreload, err := a.ldPreloadFileInstrument.mutate(ctx)
-	if err != nil {
-		return err
-	}
-	defer func() {
-		if err != nil && rollbackLDPreload != nil {
-			if err := rollbackLDPreload(); err != nil {
-				log.Warnf("Failed to rollback agent config: %v", err)
-			}
+	envs := env.FromEnv()
+	if shouldInstrumentHost(envs) {
+		rollbackLDPreload, err := a.ldPreloadFileInstrument.mutate(ctx)
+		if err != nil {
+			return err
 		}
-	}()
+		defer func() {
+			if retErr != nil && rollbackLDPreload != nil {
+				if err := rollbackLDPreload(); err != nil {
+					log.Warnf("Failed to rollback agent config: %v", err)
+				}
+			}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 Thought:** ‏I think it's less footgun-y for telemetry to keep `err` as the main error and user another name for rollback errors.

Reply:
you want to trigger a rollback on the return error when the defer func is called. To ensure that err does not get overiden/ scoped to a { }, the best way is to use a different err for return